src/rbr/Region.java

```diff
@@ -3,20 +3,20 @@
 import java.util.*;
 import util.*;
 
-public class Region {
-    private Set<Character> inputPorts;
-    private Set<Character> outputPorts;
-    private Range box;
-    private Set<Vertex> destinations;
+public final class Region {
+    private final Set<Character> inputPorts;
+    private final Set<Character> outputPorts;
+    private final Range box;
+    private final Set<Vertex> destinations;
 
     public Region(Set<Character> ip, Set<Vertex> destinations, Set<Character> op) {
         this.destinations = new HashSet<>(destinations);
         this.inputPorts = new HashSet<>(ip);
         this.outputPorts = new HashSet<>(op);
-        this.updateBox();
+        this.box = createBox();
```

Comment:
This method return something. Do not ask it to do something, instead ask for something. You could pass the destinations to make more sense.